From c8bc47393693d6d6641bfcf9343b1a9adf83be2b Mon Sep 17 00:00:00 2001
From: Federico Vaga <federico.vaga@vaga.pv.it>
Date: Sun, 23 Apr 2017 12:22:56 +0200
Subject: [PATCH 132/134] trace-cmd:read: BUG initialize input_files item to
 zero

On allocation the data structure was not initialized. Later on some
attribute of this structure are used (e.g. tsoffset) assuming that the
default value is zero, but it is not always true.

Link: http://lkml.kernel.org/r/20170423102258.21609-4-federico.vaga@vaga.pv.it

Signed-off-by: Federico Vaga <federico.vaga@vaga.pv.it>
Signed-off-by: Steven Rostedt (VMware) <rostedt@goodmis.org>
---
 trace-read.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/trace-read.c b/trace-read.c
index c8a5b59d88aa..2d48646e5d34 100644
--- a/trace-read.c
+++ b/trace-read.c
@@ -294,6 +294,7 @@ static void add_input(const char *file)
 	item = malloc(sizeof(*item));
 	if (!item)
 		die("Failed to allocate for %s", file);
+	memset(item, 0, sizeof(*item));
 	item->file = file;
 	list_add_tail(&item->list, &input_files);
 	last_input_file = item;
-- 
2.5.5

